rust/src/ldap/detect.rs

```diff
@@ -181,49 +181,67 @@ unsafe extern "C" fn ldap_detect_responses_operation_setup(
     return 0;
 }
 
-unsafe extern "C" fn ldap_detect_responses_operation_match(
-    _de: *mut c_void, _f: *mut c_void, _flags: u8, _state: *mut c_void, tx: *mut c_void,
-    _sig: *const c_void, ctx: *const c_void,
+fn match_at_index<T>(
+    tx: &LdapTransaction, ctx_value: &DetectUintData<T>,
```

Comment:
Cool, Could we have a more generic type than `LdapTransaction` to fit other protocols ?